libs/dfi/gtest/src/dyn_avpr_function_tests.cpp

```diff
@@ -219,27 +219,32 @@ const char* theAvprFile = "{ \
                 }\
             }";
 
-TEST_GROUP(DynAvprFunctionTests) {
-    void setup() override {
+
+class DynAvprFunctionTests : public ::testing::Test {
+public:
+    DynAvprFunctionTests() {
         int lvl = 1;
         dynAvprFunction_logSetup(stdLog, nullptr, lvl);
         dynAvprType_logSetup(stdLog, nullptr, lvl);
     }
+    ~DynAvprFunctionTests() override {
+    }
+
 };
 
 // Test 1, simple function with three arguments and a return type
 static int avpr_example1(__attribute__((unused)) void* handle, int32_t a, int32_t b, int32_t c, int32_t * out) {
-    CHECK_EQUAL(2, a);
-    CHECK_EQUAL(4, b);
-    CHECK_EQUAL(8, c);
+    EXPECT_EQ(2, a);
```

Comment:
Sometimes I see EXPECT_xxx and sometimes ASSERT_xxx, what is the preferred variant?

Reply:
A google test ASSERT_ is like cpputest CHECK_ . An failed EXPECT_ will fail the result of the test, but will continue running the rest of the test.
The reason why sometimes I replace CHECK_ with EXPECT instead of ASSERT_ is that ASSERT_ are not always allowed in "plain" c functions (compile error). If I am correct it has to do with exceptions.